submitit/auto/auto.py

```diff
@@ -141,6 +141,8 @@ def _internal_update_parameters(self, **kwargs: Any) -> None:
         # check type of replaced variables
         generics = AutoExecutor._valid_parameters()
         for name in generics:
```

Comment:
You're only removing generic parameters that are None.
Shouldn't we remove all parameters that are None ?
`kwargs = {k: v for k, v in kwargs.items() if v is not None}`

Reply:
The other do not matter that much, because they are not type checked as the generic parameters, so they could be handled in different ways by the actual generator.
Then again, either way is fine for me

Reply:
I think it's better to be consistent and to always treat None as not set.